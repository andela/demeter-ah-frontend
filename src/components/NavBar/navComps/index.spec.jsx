import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { AuthNav, CreateArticleNav } from './index';

describe('NavBar header Component', () => {
  it('Should render without errors', () => {
    const history = {
      location: {
        pathname: '/signup'
      }
    };
    const component = shallow(<AuthNav history={history} />);
    expect(component).toMatchSnapshot();
  });

  it('should click btn', () => {
    const history = {
      push: e => e,
    };
    const mockOnSignUpFn = jest.fn();
    const component = mount(
      <Router>
        <AuthNav history={history} />
      </Router>
    );
    const submitBtn = component.find('button').at(0);
    const submitBtn2 = component.find('button').at(1);
    submitBtn.simulate('click',
      { preventDefault() { } });
    submitBtn2.simulate('click',
      { preventDefault() { } });
    expect(mockOnSignUpFn.mock.calls.length).toBe(0);
  });

  it('should click btn', () => {
    const mockOnSignUpFn = jest.fn();
    const component = mount(
      <Router>
        <CreateArticleNav openModal={mockOnSignUpFn} />
      </Router>
    );
    const submitBtn = component.find('button').at(0);
    submitBtn.simulate('click',
      { preventDefault() { } });
    expect(mockOnSignUpFn.mock.calls.length).toBe(1);
  });
});
