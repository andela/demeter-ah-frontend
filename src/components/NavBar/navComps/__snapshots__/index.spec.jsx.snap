// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar header Component Should render without errors 1`] = `
<Fragment>
  <Link
    to="/signup"
  >
    <Button
      classes="btn-purple w-32 hidden md:inline"
      name="Sign Up"
      type="button"
    />
  </Link>
  <Link
    to="/signin"
  >
    <Button
      classes=" cursor-pointer btn-white w-32 ml-2 text-purple-650 border hidden md:inline border-solid border-purple-650"
      name="Sign In"
      type="button"
    />
  </Link>
</Fragment>
`;
